tests/cpp/core/ScreenLockTests.cpp

```diff
@@ -0,0 +1,109 @@
+/*    (INCLUDING NEGLIGENCE OR OTHERWISE)  ARISING IN ANY WAY OUT    */
+/*    OF  THE  USE OF  THIS  SOFTWARE,  EVEN  IF ADVISED  OF  THE    */
+/*    POSSIBILITY OF SUCH DAMAGE.                                    */
+/*                                                                   */
+/* The views and conclusions contained in the software and           */
+/* documentation are those of the authors and should not be          */
+/* interpreted as representing official policies, either expressed   */
+/* or implied, of Ecole polytechnique federale de Lausanne.          */
+/*********************************************************************/
+
+#define BOOST_TEST_MODULE ScreenLockTest
+
+#include <boost/test/unit_test.hpp>
+
+#include "ScreenLock.h"
+
+#include <QObject>
+
+namespace
+{
+const QString STREAM1("Stream1");
+const QString STREAM2("Stream2");
+}
+
+BOOST_AUTO_TEST_CASE(test_locking)
+{
+    auto screenLock = ScreenLock::create();
+    bool locked = false;
+    bool listChanged = false;
+    bool lockModified = false;
+    bool streamAccepted = false;
+    bool streamRejected = false;
+
+    QObject::connect(screenLock.get(), &ScreenLock::lockChanged,
+                     [&locked](bool lockState) { locked = lockState; });
+
+    QObject::connect(screenLock.get(), &ScreenLock::streamListChanged,
+                     [&listChanged]() { listChanged = true; });
+
+    QObject::connect(screenLock.get(), &ScreenLock::modified,
+                     [&lockModified]() { lockModified = true; });
+
+    QObject::connect(screenLock.get(), &ScreenLock::streamAccepted,
+                     [&streamAccepted](QString) { streamAccepted = true; });
+
+    QObject::connect(screenLock.get(), &ScreenLock::streamRejected,
+                     [&streamRejected](QString) { streamRejected = true; });
+
+    screenLock->lock();
+    BOOST_CHECK_EQUAL(locked, true);
+    BOOST_CHECK_EQUAL(lockModified, true);
+
+    screenLock->requestStreamAcceptance(STREAM1);
+    BOOST_CHECK_EQUAL(listChanged, true);
+    BOOST_CHECK(screenLock->getPendingStreams().length() == 1);
+
+    screenLock->acceptStream(STREAM1);
+    BOOST_CHECK_EQUAL(streamAccepted, true);
+    BOOST_CHECK(screenLock->getPendingStreams().length() == 0);
+
+    screenLock->requestStreamAcceptance(STREAM1);
+    BOOST_CHECK(screenLock->getPendingStreams().length() == 1);
+
+    screenLock->rejectStream(STREAM1);
+    BOOST_CHECK_EQUAL(streamRejected, true);
+    BOOST_CHECK(screenLock->getPendingStreams().length() == 0);
+
+    screenLock->requestStreamAcceptance(STREAM1);
+    BOOST_CHECK(screenLock->getPendingStreams().length() == 1);
+    screenLock->cancelStreamAcceptance(STREAM1);
+    BOOST_CHECK(screenLock->getPendingStreams().length() == 0);
+
+    screenLock->requestStreamAcceptance(STREAM1);
+    screenLock->requestStreamAcceptance(STREAM2);
+    BOOST_CHECK(screenLock->getPendingStreams().length() == 2);
+
+    screenLock->unlock();
```

Comment:
also check that calling unlock() emits streamAccepted() for each of the two pending streams